pytomorrowio/pytomorrowio.py

```diff
@@ -367,35 +365,27 @@ async def realtime_and_all_forecasts(
 
         forecasts: Dict[str, List[Dict[str, Any]]] = {}
         if all_forecasts_fields is not None:
-            forecasts = await TomorrowioV4.all_forecasts(
-                self,
+            forecasts = await self.all_forecasts(
                 all_forecasts_fields,
                 nowcast_timestep=nowcast_timestep,
                 reset_num_api_requests=False,
             )
         else:
-            if nowcast_fields is not None:
-                forecasts[NOWCAST] = await TomorrowioV4.forecast_nowcast(
-                    self,
-                    nowcast_fields,
-                    timestep=nowcast_timestep,
-                    reset_num_api_requests=False,
-                )
-            for fields, forecast_type, method in (
-                (hourly_fields, HOURLY, TomorrowioV4.forecast_hourly),
-                (daily_fields, DAILY, TomorrowioV4.forecast_daily),
-            ):
-                if fields is not None:
-                    forecasts[forecast_type] = await method(
-                        self, fields, reset_num_api_requests=False
+            for fields, timestep in [
+                (nowcast_fields, timedelta(minutes=nowcast_timestep)),
+                (hourly_fields, ONE_HOUR),
+                (daily_fields, ONE_DAY),
+            ]:
+                if fields:
+                    if forecasts:
+                        await asyncio.sleep(1)
```

Comment:
This function could generate up two four calls in rapid succession could exceed the 3 TPS allowed for free accounts.

Reply:
good point, I had a sleep in here before but removed it. Let's get the other PR merged in then we can figure out how to handle this

Reply:
I think the better way to handle this is to add a sleep in `_call_api` if we have only one call remaining in the second

Reply:
https://github.com/raman325/pytomorrowio/pull/15

Reply:
I like the idea of using the headers. We'd need to keep track of when the headers were received to know if they're still relevant. And all bets are off if multiple instances are sharing the API key.

Reply:
I think we can store the headers in HA and implement wait logic in the coordinator if it's really a concern